Mathlib/CategoryTheory/Sites/Coherent/ExtensiveSheaves.lean

```diff
@@ -88,10 +89,32 @@ theorem isSheaf_iff_preservesFiniteProducts [FinitaryExtensive C] (F : Cᵒᵖ 
       ext
       simp
   · let _ := hF.some
-    rw [Presieve.isSheaf_coverage]
+    erw [Presieve.isSheaf_coverage]
     intro X R ⟨Y, α, Z, π, hR, hi⟩
     have : IsIso (Sigma.desc (Cofan.inj (Cofan.mk X π))) := hi
     have : R.Extensive := ⟨Y, α, Z, π, hR, ⟨Cofan.isColimitOfIsIsoSigmaDesc (Cofan.mk X π)⟩⟩
     exact isSheafFor_extensive_of_preservesFiniteProducts R F
 
+/--
+A presheaf on a category which is `FinitaryExtensive` is a sheaf iff it preserves finite products.
+-/
+theorem Presheaf.isSheaf_iff_preservesFiniteProducts {D : Type*} [Category D]
+    [FinitaryExtensive C] (F : Cᵒᵖ ⥤ D) :
+    IsSheaf (extensiveTopology C) F ↔ Nonempty (PreservesFiniteProducts F) := by
+  constructor
+  · intro h
+    rw [IsSheaf] at h
+    refine ⟨⟨fun J _ ↦ ⟨@fun K ↦ ⟨@fun c hc ↦ ?_⟩⟩⟩⟩
```

Comment:
Instead of `@fun`, you may favour the `fun {x}` syntax.

Reply:
Ah thanks, that's nicer!